fxrecorder/src/bin/main.rs

```diff
@@ -29,15 +33,39 @@ fn main() {
 }
 
 async fn fxrecorder(log: Logger, _options: Options, config: Config) -> Result<(), Box<dyn Error>> {
-    use libfxrecorder::proto::RecorderProto;
-    use tokio::net::TcpStream;
+    {
+        let stream = TcpStream::connect(&config.host).await?;
+        info!(log, "Connected"; "peer" => config.host);
 
-    let stream = TcpStream::connect(&config.host).await?;
-    info!(log, "Connected"; "peer" => config.host);
+        let mut proto = RecorderProto::new(log.clone(), stream);
 
-    let mut proto = RecorderProto::new(log, stream);
+        proto.handshake(true).await?;
+    }
+
+    {
+        let reconnect = || {
+            info!(log, "Attempting re-connection to runner...");
+            TcpStream::connect(&config.host)
+        };
+
+        // 30 * 2^4 = 7:30
```

Comment:
Nit: isn't `30 * 2^4 == 8:00`? I think this actually *will* wait 7:30, because it should be `30 * 2^0 + 30 * 2^1 + 30 * 2^2 + 30 * 2^3 == 30 * (2^4 - 1) == 7:30`, but I think the comment is just slightly wrong, no?

Reply:
Yeah this is a sloppy comment.